library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -710,11 +711,23 @@ public void emitTuples()
   }
 
   /**
+   * Notifies that the directory is being scanned.<br>
+   * Override this method to custom handling. Will be called once
+   */
+  protected void visitDirectory(Path filePath)
+  {
+  }
+
+  /**
    * Scans the directory for new files.
    */
   protected void scanDirectory()
   {
     if (System.currentTimeMillis() - scanIntervalMillis >= lastScanMillis) {
+      if (!visitedDirectory) {
```

Comment:
This will not take care of empty child directory. e.g. you had a directory with following structure
a
|
1.txt, 2.txt, c

and dir "c" is empty  we need to send a control tuple for "c" as well.